content/python/pagination.py.md

```diff
@@ -1,52 +1,143 @@
 ---
-title: "pagination.py"
-description: "pagination.py"
-date: "2017-11-23"
+title: "Pagination Examples"
+description: "A collection of examples on how to properly get large listings of objects from the SoftLayer API using Pagination,
+    Python iterators, and objectFilters."
+date: "2021-11-12"
 classes: 
     - "SoftLayer_Account"
 tags:
     - "pagination"
 ---
 
+When querying the SoftLayer API and using a resultLimit to paginate your results, you should almost always use an `OrderBy` [objectFilter](/article/object-filters/) to sort your results, as by default most results will not have an order imposed on them by the database. It is fairly common for the database to have differeing orders between queries UNLESS an `OrderBy` filter is used.
+
+The python client itself has a built in function called [`iter_call`](https://softlayer-python.readthedocs.io/en/latest/api/client/#SoftLayer.API.BaseClient.iter_call) which will return a python [generator](https://wiki.python.org/moin/Generators). This generator will only get the next set of results when you need them, which can make doing operations on large datasets feel faster in that you don't have to wait for the entire dataset to be retrieved before obtaining some results.
 
-```
-"""
-Example about pagination
 
-The script retrieves the Bare metal servers in a account
-and displays all the servers using pagination
+## [The Setup](#the-setup) {#the-setup .anchor-link}
 
-Important manual pages
-http://sldn.softlayer.com/reference/services/SoftLayer_Account
-http://sldn.softlayer.com/reference/services/SoftLayer_Account/getHardware
-https://softlayer-api-python-client.readthedocs.org/en/latest/api/client/#module-SoftLayer
+To avoid having to type the boiler-plate code a lot, assume each of these example function will be a member function of the following `Paginator` class
 
-License: http://sldn.softlayer.com/article/License
-Author: SoftLayer Technologies, Inc. <sldn@softlayer.com>
-"""
+```python
 import SoftLayer
 
-# Your SoftLayer API username and key.
-USERNAME = 'set me'
-API_KEY = 'set me'
-
-client = SoftLayer.Client(username=USERNAME, api_key=API_KEY)
-
-offset = 0
-limit = 1
-
-accountService = client['SoftLayer_Account']
-
-while True:
-    try:
-        result = accountService.getHardware(limit=limit, offset=offset)
-        offset = offset + limit
-        limit = limit + limit
-        print(result)
-        if not result:
-            break
-    except SoftLayer.SoftLayerAPIError as e:
-        print("Unable to retrieve the servers . " % (e.faultCode, e.faultString))
-    exit(1)
-    
+class Paginator():
```

Comment:
the name is funny

Reply:
hopefully not funny in a bad way lol.  After looking this up, it seems `Paginator` might not be an actual real word (at least its not in [the dictionary](https://www.merriam-webster.com/dictionary/paginator) ), but it has some common uses for "something that does pagination".